backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/impexp/spreadsheet/csv/CSVWriter.java

```diff
@@ -130,8 +134,9 @@ public void appendHeaderIfNeeded()
 				headerLine.append(fieldDelimiter);
 			}
 
-			final String headerColQuoted = quoteCsvValue(headerCol);
-			headerLine.append(headerColQuoted);
+			final String headerColEffective = doNotQuoteRows ? removeQuotesFromCsvValue(headerCol) : quoteCsvValue(headerCol);
```

Comment:
i suggest to refactor `doNotQuoteRows. removeQuotesFromCsvValue(headerCol) : quoteCsvValue(headerCol);` to a new method, e.g. formaCsvCell(final String string).
You shall use it in toCsvValue too.

Next, i would check exactly what 
* escapeSpecialCharacters
* removeQuotesFromCsvValue
do.

Maybe you can consider merging all together in formaCsvCell

Reply:
regarding the escapeSpecialCharacters part, i.e.
> valueStr.contains(fieldDelimiter) ? quoteCsvValue(valueStr) : valueStr;

So to understand... in case our string contains the field delimiter, we will quote no matter what... right?

Reply:
ahh, and now that i am thinking again... i think `toCsvValue` shall be the function to be used and improved.
No need for a new `formatCsvCell` function. wdyt?